### PR TITLE
[SELECT][Week 2][Assignment submission] Movie Viewer

### PR DESCRIPTION
**Pull request by huydangquoc**
Hi @chug2k

My assignment was submitted via coderschool website.
As per your request, I create this pull request for your convenience.
Sorry for ugly branching :-) It's my workaround in order to create pull request.
Please help review.

Best Regards,
Huy Dang

### CODE REVIEW

File: MovieViewer/Movie.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//  Movie.swift
+//  MovieViewer
+//
+//  Created by Dang Quoc Huy on 6/30/16.
+//  Copyright © 2016 Dang Quoc Huy. All rights reserved.
+//
+
+import Foundation
+
+struct Movie {
```
Comment (chug2k) [inline MovieViewer/Movie.swift:11]: Good job using a struct!

File: MovieViewer/Movie.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    
+    var title: String?
+    var overview: String?
+    var posterPath: String?
```
Comment (chug2k) [inline MovieViewer/Movie.swift:15]: consider using a getter if you need multiple poster paths

```
var thumbnailPath: NSURL {
  return ....
}
```

File: MovieViewer/MovieDetailViewController.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    @IBOutlet weak var detailViewHeightConstrain: NSLayoutConstraint!
+    @IBOutlet weak var contentViewHeightConstrain: NSLayoutConstraint!
+    
+    var movie: Movie?
+    
+    // Called after the controller's view is loaded into memory
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        titleLabel.text = movie!.title
```
Comment (chug2k) [inline MovieViewer/MovieDetailViewController.swift]: I would wrap this as ...

```
guard let movie = movie else { return }
```

File: MovieViewer/MovieDetailViewController.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        overviewLabel.text = movie!.overview
+        releaseDateLabel.text = movie!.releaseDate
+        
+        let thumbnail = TMDBClient.BaseImageW154Url + movie!.posterPath!
```
Comment (chug2k) [inline MovieViewer/MovieDetailViewController.swift]: Also please use a guard or if let on posterPath

File: MovieViewer/MoviesViewController.swift [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+//
+//  Created by Dang Quoc Huy on 6/30/16.
+//  Copyright © 2016 Dang Quoc Huy. All rights reserved.
+//
+
+import UIKit
+import AFNetworking
+import MBProgressHUD
+
+enum MoviesViewMode {
```
Comment (chug2k) [inline MovieViewer/MoviesViewController.swift:13]: great usage of enum type! :smile:

File: MovieViewer/UIImageView.swift [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+//
+
+import UIKit
+
+extension UIImageView {
+    
+    // Set image with fade in animation
+    func setImageWithFadeIn(image: UIImage) {
+        
+        self.alpha = 0.0
```
Comment (chug2k) [inline MovieViewer/UIImageView.swift]: do you need `self` here?

**Comment by huydangquoc** [comment]
Hi @chung2k

Thank you for your review and comments.
Code is updated accordingly and merged to master

Best Regards,
Huy Dang

**Pull request merged**